### PR TITLE
Fix plugin linking in mosviz

### PR DESCRIPTION
**Pull request by javerbukh**
### Description

Using model fitting in mosviz does not link the model to the spectrum 1d reference data. This line explicitly calls the linking method and links the reference data to the model.

EDIT: This PR was updating to fix all mosviz plugin linking.

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [ ] Are two approvals required? Branch protection rule does not check for the second approval. If a second approval is not necessary, please apply the `trivial` label.
- [ ] Do the proposed changes actually accomplish desired goals? Also manually run the affected example notebooks, if necessary.
- [ ] Do the proposed changes follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Are tests added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Are docs added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Did the CI pass? If not, are the failures related?
- [ ] Is a change log needed? If yes, is it added to `CHANGES.rst`?
- [ ] Is a milestone set?
- [ ] After merge, any internal documentations need updating (e.g., JIRA, Innerspace)?

### CODE REVIEW

**Comment by pllim** [comment]
I think we need to add a test too, if possible.

**Comment by javerbukh** [comment]
#1412 links newly added data to the data in `data_collection[0]`, which in mosviz in the table data. I wanted to make the logic more sophisticated in app.py for adding new data from mosviz but I wasn't sure if I had the time. I can certainly add a test.

**Comment by kecnry** [comment]
Does #1451 and https://github.com/spacetelescope/jdaviz/pull/1452#pullrequestreview-1029195380 suggest that a more general approach might be needed for all plugins in mosviz to link correctly?  (Or if general is difficult, then at least copying this line to all mosviz plugins?)

File: jdaviz/configs/default/plugins/model_fitting/model_fitting.py [hunk 1]
```diff
@@ -576,3 +576,7 @@ def vue_register_spectrum(self, event):
 
         self.add_results.add_results_from_plugin(spectrum)
         self._set_default_results_label()
+
+        # Link the result spectrum to the reference data of the spectrum viewer
+        self.app._link_new_data(self.app.get_viewer('spectrum-viewer').state.reference_data.label,
+                                self.results_label)
```
Comment (kecnry) [inline jdaviz/configs/default/plugins/model_fitting/model_fitting.py:582]: is this redundant now with the logic in `app.py` or does it need to stay in?

Comment (javerbukh) [reply jdaviz/configs/default/plugins/model_fitting/model_fitting.py:582]: Ah, good catch!

**Review by kecnry — APPROVED** [review]
Seems to do the trick, thanks!

**Comment by javerbukh** [comment]
Follow up ticket #1459

**Review by rosteen — APPROVED** [review]
Approving, since the problems I see are also present on `main` and there's a follow-up ticket for tests.

**Pull request merged**